pkg/config/envprovider/provider_test.go

```diff
@@ -0,0 +1,78 @@
+package envprovider
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.signoz.io/signoz/pkg/config"
+)
+
+func TestGetWithStrings(t *testing.T) {
+	t.Setenv("SIGNOZ_K1_K2", "string")
+	t.Setenv("SIGNOZ_K3__K4", "string")
+	t.Setenv("SIGNOZ_K5__K6_K7__K8", "string")
+	t.Setenv("SIGNOZ_K9___K10", "string")
+	t.Setenv("SIGNOZ_K11____K12", "string")
```

Comment:
anything that doesn't contain signoz prefix shouldn't be considered, prove using test.